src/rendergraph/common/rendergraph/attributeset.h

```diff
@@ -4,17 +4,17 @@
 #include "rendergraph/attributeinit.h"
 
 namespace rendergraph {
-class AttributeSet;
-}
 
-class rendergraph::AttributeSet : public rendergraph::BaseAttributeSet {
+class AttributeSet : public BaseAttributeSet {
   public:
     AttributeSet(std::initializer_list<AttributeInit> list, const std::vector<QString>& names);
 };
 
-namespace rendergraph {
-template<typename... T>
-AttributeSet makeAttributeSet(const std::vector<QString>& names) {
-    return AttributeSet({(AttributeInit::create<T>())...}, names);
+template<typename... Ts, int N>
+AttributeSet makeAttributeSet(const QString (&names)[N]) {
+    static_assert(sizeof...(Ts) == N);
+    return AttributeSet({(AttributeInit::create<Ts>())...},
+            std::vector<QString>(std::cbegin(names), std::cend(names)));
```

Comment:
we might want to consider propagating this further, because the constructor also just copies stuff out of the vector to assemble another vector. So this vector is conceptually still just another temporary allocation.

Reply:
Do we really care? This isn't exactly in the critical path!

Reply:
probably not. I'll look into though and see if its worth the extra code.